src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpKeywordHighlighting.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpKeywordHighlighting : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpKeywordHighlighting(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpKeywordHighlighting))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public void Foreach()
+        {
+            VisualStudio.Editor.SetText(@"
+class C
+{
+    void M()
+    {
+        foreach(var c in """") { if(true) break; else continue; }
+    }
+}");
+
+            Verify("foreach", 3);
```

Comment:
These tests are strange. I'd expect something with markup indicating which spans are highlighted and checking that.